pkg/proxy/ipvs/netlink_linux.go

```diff
@@ -187,3 +175,51 @@ func (h *netlinkHandle) GetLocalAddresses(dev, filterDev string) (sets.String, e
 	}
 	return res, nil
 }
+
+// ChangeLORouteSrc change the src of lo link in table local, it's equivalent to exec:
+// ip route show table local dev lo > result
+// ip route change <result> src <src>
+func (h *netlinkHandle) ChangeLORouteSrc(src string) error {
+	lo, err := h.LinkByName("lo")
+	if err != nil {
+		return fmt.Errorf("get lo link err: %v", err)
+	}
+
+	routes, err := h.listRoutesForLink(lo.Attrs().Index)
+	if err != nil {
+		return fmt.Errorf("list routes for link lo err: %v", err)
+	}
+	if len(routes) != 0 {
+		srcIP := net.ParseIP(src)
+		for _, r := range routes {
+			route := netlink.Route{
+				Dst:       r.Dst,
+				Table:     r.Table,
+				Src:       srcIP,
+				LinkIndex: r.LinkIndex,
+				Protocol:  r.Protocol,
+				Scope:     r.Scope,
+			}
+			if err := h.RouteReplace(&route); err != nil {
+				return fmt.Errorf("change route err: change to: %v, err: %v", route, err)
+			}
+		}
+	}
+	return nil
+}
+
+func (h *netlinkHandle) listRoutesForLink(linkIndex int) ([]netlink.Route, error) {
+	routeFilter := &netlink.Route{
+		Table:    unix.RT_TABLE_LOCAL,
+		Type:     unix.RTN_LOCAL,
+		Protocol: unix.RTPROT_KERNEL,
```

Comment:
I think we could add: "Scope: unix.RT_SCOPE_HOST" to the filter

Reply:
Actually we could probably also filter on destination here (127.0.0.1/32 and maybe 127.0.0.1/8)

Reply:
I think `lindIndex` filter for `lo` would be enough?

Reply:
`lo` device will also include broadcast routes but yes if you also filter on `host` scope you will only get the proper routes